### PR TITLE
SNOW-1733956: fix sql count of modin tests with cte enabled

### PR DESCRIPTION
**Pull request by sfc-gh-jdu**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-NNNNNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [ ] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Review by sfc-gh-aalam — APPROVED** [review]
if this change also affects daily precommit, can you post a successful run for those as well

File: .github/workflows/daily_modin_precommit.yml [hunk 1]
```diff
@@ -242,8 +242,8 @@ jobs:
             .tox/.coverage
             .tox/coverage.xml
 
-  test-enable-cte-optimization:
-    name: Test Enable CTE Optimization modin-${{ matrix.os.download_name }}-${{ matrix.python-version }}-${{ matrix.cloud-provider }}
+  test-disable-cte-optimization:
```
Comment (sfc-gh-helmeleegy) [inline .github/workflows/daily_modin_precommit.yml:245]: I'm not sure I'm getting it. How come we're disabling the CTE optimization here and inn many other places, but still decreasing the sql counts?

Comment (sfc-gh-jdu) [reply .github/workflows/daily_modin_precommit.yml:245]: This is a new test with disable cte optimization. In this test, we actually ignore sql count check (https://github.com/snowflakedb/snowpark-python/blob/37fe60335c17e0594831bb14632c7391fd9561f7/.github/workflows/daily_modin_precommit.yml#L298)

**Review by sfc-gh-helmeleegy — APPROVED** [review]

**Review by sfc-gh-nkumar — APPROVED** [review]
This is great!!

**Review by sfc-gh-yzou — APPROVED** [review]

**Comment by sfc-gh-jdu** [comment]
> if this change also affects daily precommit, can you post a successful run for those as well

yes, https://github.com/snowflakedb/snowpark-python/actions/runs/11485902072, though some tests seem failing with an unrelated error:

[31mFAILED[0m tests/integ/modin/io/test_read_csv.py::[1mtest_read_csv_usecols_nonexistent_negative[usecols1][0m - snowflake.connector.errors.ProgrammingError: 000612 (02000): Result not found

**Pull request merged**